peewee.py

```diff
@@ -2437,15 +2446,31 @@ def filter(self, *args, **kwargs):
 
         dq_node = dq_node.rhs
 
+        seen = set()
+        remove_dupes = lambda lst: seen.clear() or [
+            i for i in lst if i not in seen and not seen.add(i)]
```

Comment:
It clears the set before returning the result of the list comprehension. The returned list does not contain any duplicate items. I could move this to a separate method and out of a list comprehension.

http://stackoverflow.com/questions/7961363/python-removing-duplicates-in-lists